src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingFlorist.java

```diff
@@ -149,15 +150,14 @@ public void removePlantableGround(final BlockPos pos)
     public ItemStack getFlowerToGrow()
     {
         final List<ItemStorage> stacks = getPlantablesForBuildingLevel(getBuildingLevel()).stream()
-                                           .filter(stack -> !getModuleMatching(ItemListModule.class, m -> m.getId().equals(BUILDING_FLOWER_LIST)).isItemInList(stack))
-                                           .collect(Collectors.toList());
+          .filter(stack -> !getModuleMatching(ItemListModule.class, m -> m.getId().equals(BUILDING_FLOWER_LIST)).isItemInList(stack)).toList();
+
         if (stacks.isEmpty())
         {
             return null;
```

Comment:
No, turn this into an Empty stack, or make the method return an optional!

Reply:
I'm not changing the logic here, just a hyper expensive shuffle call.

Reply:
Make a Todo out of it then
Null itemstacks are forbidden you know that

Reply:
If the method is nullable then it's fine, I don't see the necessity to handle around optionals and having to unpack them constantly. the null here indicates that sth is wrong anyway,